tests/unit/amo/components/TestSearchForm.js

```diff
@@ -1,179 +1,386 @@
+      wrapper.find('input').simulate('change', createFakeChangeEvent());
+      expect(dispatch.calledOnce).toBe(true);
+      expect(dispatch.calledWith(autocompleteCancel())).toBe(true);
+    });
+
+    it('displays suggestions when user is typing', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
 
-  it('passes the context through', () => {
-    const { store } = dispatchSignInActions();
-    store.dispatch(setViewContext(ADDON_TYPE_EXTENSION));
+      wrapper = mountComponent({
+        query: 'foo',
+        suggestions: autocompleteState.suggestions,
+      });
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      // this triggers Autosuggest
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(2);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
+
+    it('does not display suggestions when search is empty', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      // setting the `query` prop to empty also sets the input state to empty.
+      wrapper = mountComponent({
+        query: '',
+        suggestions: autocompleteState.suggestions,
+      });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+    });
+
+    it('does not display suggestions when there is no suggestion', () => {
+      wrapper = mountComponent({ suggestions: [] });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
+
+    it('displays 10 loading bars when suggestions are loading', () => {
+      wrapper = mountComponent({
+        suggestions: [],
+        loadingSuggestions: true,
+      });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(LoadingText)).toHaveLength(10);
+    });
+
+    it('updates the state and push a new route when a suggestion is selected', () => {
+      const result = createFakeAutocompleteResult();
+      const { store } = dispatchAutocompleteResults({ results: [result] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      wrapper = mountComponent({
+        query: 'foo',
+        suggestions: autocompleteState.suggestions,
+      });
+      expect(wrapper.state('searchValue')).toBe('foo');
+
+      wrapper.find('input').simulate('focus');
+      wrapper.find(Suggestion).simulate('click');
+      expect(wrapper.state('searchValue')).toBe('');
+      expect(router.push.calledOnce).toBe(true);
+      expect(router.push.calledWith(result.url)).toBe(true);
```

Comment:
do a search in this file for `calledOnce` and `calledWith` so you can catch all of these